### PR TITLE
Spearbit 184

### PR DESCRIPTION
**Pull request by LayneHaber**
## Type of change

- [ ] Docs change / dependency upgrade
- [ ] Configuration / tooling changes
- [ ] Refactoring
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Requires changes in customer code

## Related Issue(s)

Fixes

### CODE REVIEW

File: packages/deployments/contracts/contracts/core/connext/facets/TokenFacet.sol [hunk 1]
```diff
@@ -398,6 +401,17 @@ contract TokenFacet is BaseConnextFacet {
     if (s.canonicalToAdopted[_key] != _adoptedAssetId || s.canonicalToRepresentation[_key] != _representation)
       revert TokenFacet__removeAssetId_invalidParams();
 
+    // Sanity check: no value custodied if on canonical domain
+    bool onCanonical = s.domain == _canonicalDomain;
+    if (onCanonical && s.custodied[_representation] > 0) {
+      revert TokenFacet__removeAssetId_remainsCustodied();
+    }
```
[the remark this answers is not in the document]
Comment (LayneHaber) [reply packages/deployments/contracts/contracts/core/connext/facets/TokenFacet.sol:408]: if you are on the canonical domain, it should be `address(0)`. in this case, we should cast from the `_canonicalId` if possible (or use reverse lookup)

[the remark this answers is not in the document]
Comment (jakekidd) [reply packages/deployments/contracts/contracts/core/connext/facets/TokenFacet.sol]: cc @LayneHaber pls check to make sure this solution covers the request/comment here

[the remark this answers is not in the document]
Comment (LayneHaber) [reply packages/deployments/contracts/contracts/core/connext/facets/TokenFacet.sol]: `representation` should be cast to `canonical` (which @jakekidd did!)

[the remark this answers is not in the document]
Comment (LayneHaber) [reply packages/deployments/contracts/contracts/core/connext/facets/TokenFacet.sol]: verified fixes, and added more tests

**Comment by LayneHaber** [comment]
note: tests working; forge install failing

**Pull request merged**